Change documentation for "default" tag in map configuration (#1282)
hazelcast/hz-docs: docs/modules/data-structures/pages/map-config.adoc

For details on in-memory format, refer to xref:setting-data-format.adoc[].

== The Default (Fallback) Map Configuration

> When a map is created, if the map name matches an entry in hazelcast.xml/yaml file, the values in the matching entry are used to overwrite initial values
> specified in <<map-configuration-defaults,Map Configuration Defaults>> section.

Reviewer: ```suggestion
When a map is created, if the map name matches an entry in the `hazelcast.xml`/`hazelcast.yaml` file, the values in the matching entry are used to overwrite the initial values
discussed in the <<map-configuration-defaults,Map Configuration Defaults>> section.
```

> This path also includes wildcard matches, meaning entries with “*” character

Reviewer: ```suggestion
Maps that do not have any configuration defined use the default configuration. If you want to set a configuration that is valid for all maps, you can name your configuration as `default`. A user-defined default configuration applies to every map that does not have a specific custom map configuration defined with the map’s name. You can also use wildcards to associate your configuration with multiple map names. See the [configuration documentation](https://docs.hazelcast.com/hazelcast/5.5/configuration/using-wildcards) for more information about wildcards.
```

This is actually a snippet taken from Cloud's docs, and I think it's pretty well worded: https://docs.hazelcast.com/cloud/map-configurations

> If the map name does not match any entry in the hazelcast.xml/yaml file then
> 
> - if “default” map configuration exists, the values under this tag are used to overwrite initial values. Therefore, "default" serves as a fallback.
> 
> - if a “default” map configuration does not exist, the map is created with initial values

Reviewer: ```suggestion
When a map name does not match any entry in the `hazelcast.xml`/`hazelcast.yaml` file then:

- If the `default` map configuration exists, the values under this entry are used to overwrite initial values. Therefore, `default` serves as a fallback.

- If a `default` map configuration does not exist, the map is created with initial values as discussed in <<map-configuration-defaults,Map Configuration Defaults>>.
```
